src/contexts/AssetContext/Provider.tsx

```diff
@@ -47,18 +47,17 @@ const AssetProvider: React.FC = ({ children }) => {
 //
 
 // Takes in the asset results from a getAccountAssets() call and fetches additional asset details
-// such as name/description
-async function processAssetResults(assetsToProcess: Array<IAsset>, processingFn: (assetId: string) => Promise<false | IGetAssetResult>) {
+// such as name/description/decimals/totalSupply
+async function processAssetResults(assetsToProcess: Array<IAccountAsset>, processingFn: (assetId: string) => Promise<false | IGetAssetResult>) {
   for (const [index, asset] of assetsToProcess.entries()) {
-    // make a secondary call to get the asset's name for more friendly display
+    // call the processing function to retrieve additional details
     const result: false | IGetAssetResult = await processingFn(asset.asset);
```

Comment:
may want to add a try/catch here in case of a failure.

Reply:
good idea, done